### PR TITLE
Jac/delete extract

### PR DESCRIPTION
**Pull request by jacalata**
handle new arguments for extract jobs: --include-all, --synchronous, --timeout
list views for each workbook with --details

### CODE REVIEW

File: tabcmd/commands/extracts/extracts.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        container = Server.get_project_by_name_and_parent_path(
+            logger, server, args.project_name, args.parent_project_path
+        )
+        if args.datasource:
+            logger.debug(_("export.status").format(args.datasource))
+            datasource = Server.get_data_source_item(logger, server, args.datasource, container)
+            return datasource
+
+        elif args.workbook or args.url:
+            logger.info(_("export.status").format(args.workbook))
```
Comment (anyoung-tableau) [inline tabcmd/commands/extracts/extracts.py:21]: Should this logging be moved into the `if` and `else` blocks below? What if `args.workbook` is not set?

File: tabcmd/commands/extracts/extracts.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            if args.workbook:
+                workbook_item = Server.get_workbook_item(logger, server, args.workbook, container)
```
Comment (anyoung-tableau) [inline tabcmd/commands/extracts/extracts.py:23]: Any reason to add or not to add the `TSC.WorkbookItem` type like we do below in the `else` block?

File: tabcmd/commands/extracts/refresh_extracts_command.py [hunk 3]
```diff
@@ -34,12 +29,11 @@ def run_command(args):
         logger.debug(_("tabcmd.launching"))
         session = Session()
         server = session.create_session(args, logger)
-        refresh_action = "refresh"
 
         if args.addcalculations or args.removecalculations:
             logger.warning(
-                "Data Acceleration tasks are deprecated and this parameter has no effect."
-                "It will be removed in a future update."
+                "Add Calculations tasks are deprecated and this parameter has no effect."
```
Comment (anyoung-tableau) [inline tabcmd/commands/extracts/refresh_extracts_command.py:35]: Should this string say `Add/Remove Calculations`? (from line 33)

File: tabcmd/commands/extracts/create_extracts_command.py [hunk 4]
```diff
@@ -19,37 +18,39 @@ class CreateExtracts(Server):
...
             )
-
+        )
+        try:
+            item = Extracts.get_wb_or_ds_for_extracts(args, logger, server)
             if args.datasource:
-                data_source_item = Server.get_data_source_item(logger, server, args.datasource)
                 logger.info(_("createextracts.for.datasource").format(args.datasource))
-                job = server.datasources.create_extract(data_source_item, encrypt=args.encrypt)
+                job = server.datasources.create_extract(item, encrypt=args.encrypt)
```
Comment (anyoung-tableau) [inline tabcmd/commands/extracts/create_extracts_command.py:42]: Should we also add the `TSC.JobItem` type like we do elsewhere?

File: tests/e2e/online_tests.py [hunk 5]
```diff
@@ -264,109 +266,95 @@ def test_delete_projects(self):
...
-    @pytest.mark.order(11)
-    def test_wb_delete(self):
-        name_on_server = OnlineCommandTest.TWBX_WITH_EXTRACT_NAME
-        self._delete_wb(name_on_server)
-
     @pytest.mark.order(11)
     def test_wb_publish_embedded(self):
-        name_on_server = OnlineCommandTest.TWB_WITH_EMBEDDED_CONNECTION
+        pytest.skip("waremart failed to establish a connection to your datasource.")
         file = os.path.join("tests", "assets", OnlineCommandTest.TWB_WITH_EMBEDDED_CONNECTION)
```
Comment (anyoung-tableau) [inline tests/e2e/online_tests.py:299]: Does this line need to change to `EMBEDDED_TWB_NAME` too?

File: tabcmd/commands/extracts/extracts.py (hunk 2 above)
Comment (jacalata) [reply tabcmd/commands/extracts/extracts.py:23]: None. I  just don't usually think of it for the return of 'get_workbook_item' and methods with obvious return types like that.

File: tests/e2e/online_tests.py (hunk 5 above)
Comment (jacalata) [reply tests/e2e/online_tests.py:299]: No, this line wants the filename, not the document name.

**Pull request merged**